scripts/policy/frameworks/telemetry/prometheus.zeek

```diff
@@ -2,9 +2,9 @@
 ##! Prometheus exposition and import all metrics from the
 ##! `zeek/cluster/metrics/...` topic.
 ##!
-##! For customization or disabling, redef the involved Broker options again.
-##! Specifically, to disable listening on port 9911, set
-##! :zeek:see:`Broker::metrics_port` to `0/unknown` again.
+##! For customization or disabling, redef the involved Telemetry options
+##! again. Specifically, to disable listening on port 9911, set
+##! :zeek:see:`Telemetry::metrics_port` to `0/unknown` again.
```

Comment:
I am not a fan of coupling a specific port value 0/unknown with disabling metrics functionality.

With this patch users are able to query an endpoint for the actually used port which opens up the door to let the OS assign ports which could simplify configuration; typically port 0 is used for that, mirroring the behavior of `bind`.

 It seems that the underlying endpoint impl can both [parse](https://github.com/civetweb/civetweb/blob/d7ba35bbb649209c66e582d5a0244ba988a15159/src/civetweb.c#L15486-L15501) addresses with port 0 as well as [pass that through to the OS for automatic port assignment](https://github.com/civetweb/civetweb/blob/d7ba35bbb649209c66e582d5a0244ba988a15159/src/civetweb.c#L15896) so IMO it would be unfortunate to completely close the door on even allowing that by this API choice.

Would you be open to instead introducing a dedicated flag for to toggle metrics functionality, e.g., `--enable-metrics` or `--disable-metrics`?

Reply:
An intermediate solution could be to sharpen the interpretation of "port 0" when parsing the config to only interpret `0/unknown` (or maybe any port on an `unknown` protocol) as disabling, https://github.com/zeek/zeek/blob/3c811eccdbd3238e19b3282b2890327c426915e4/src/telemetry/Manager.cc#L34

With that it would be impossible to disable metrics via an env var though since it never contains a protocol.

Reply:
> It seems that the underlying endpoint impl can both [parse](https://github.com/civetweb/civetweb/blob/d7ba35bbb649209c66e582d5a0244ba988a15159/src/civetweb.c#L15486-L15501) addresses with port 0 as well as [pass that through to the OS for automatic port assignment](https://github.com/civetweb/civetweb/blob/d7ba35bbb649209c66e582d5a0244ba988a15159/src/civetweb.c#L15896) so IMO it would be unfortunate to completely close the door on even allowing that by this API choice.

We don't notify the user in any way what port number is being used by the telemetry manager, since we expect them to set it themselves as part of the configuration. If CivetWeb picks its own port, the user won't know what that port is, nor will the services.json endpoint know what any of the ports are. I'm inclined to not allow it to be configured like that.

> Would you be open to instead introducing a dedicated flag for to toggle metrics functionality, e.g., --enable-metrics or --disable-metrics?

How about a script-level setting that overrides any other ports you set? That would cover cluster mode.

> With that it would be impossible to disable metrics via the ZEEK_METRICS_PORT env var though since it never contains a protocol, only a port number.

In standalone mode, not setting `ZEEK_METRICS_PORT` and not setting `Telemetry::metrics_port` already disables creating the Prometheus server.

Reply:
I ended up removing all of the environment variables, in favor of controlling everything from script-land instead.